git-hooks/pre-commit

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Get a list of changed Go files
+GO_FILES=$(git diff --cached --name-only --diff-filter=ACM | grep '\.go$')
+
+# If no Go files are changed, skip the rest of the hook
+if [ -z "$GO_FILES" ]; then
+    echo "No Go files to process."
+    exit 0
+fi
+
+# 1. Run gofumpt in check mode to verify formatting
+echo "Checking gofumpt formatting..."
+gofumpt -l . | grep -q . && {
```

Comment:
this is no bueno. we don't want to force the whole repo to be formatted, we should only look at changes files here